src/box/field_default_func.c

```diff
@@ -0,0 +1,134 @@
+/*
+ * SPDX-License-Identifier: BSD-2-Clause
+ *
+ * Copyright 2010-2023, Tarantool AUTHORS, please see AUTHORS file.
+ */
+#include "port.h"
+#include "engine.h"
+#include "tt_static.h"
+#include "tuple_format.h"
+#include "field_default_func.h"
+
+/**
+ * Find `default_func` in func cache.
+ * Return NULL if something went wrong (diag is set).
+ */
+struct func *
+field_default_func_find(struct field_default_func *default_func)
+{
+	struct func *func = func_by_id(default_func->id);
+	if (func == NULL) {
+		diag_set(ClientError, ER_NO_SUCH_FUNCTION,
+			 int2str(default_func->id));
+	}
+	return func;
+}
+
+/**
+ * Check `func` for compliance with the field default function rules.
+ *
+ * Returns 0 on success. On error, sets diag and returns -1.
+ * `fieldno` is used for better error message.
+ */
+static int
+field_default_func_verify(struct func *func, uint32_t fieldno)
+{
+	const char *func_name = func->def->name;
+
+	if (func->def->language == FUNC_LANGUAGE_LUA &&
```

Comment:
`!FUNC_LANGUAGE_LUA` branch isn't covered. Can you add a test that would use an SQL expression for a default value function?

Reply:
Currently it's not possible to call SQL default function, added:
```c
	/* TODO(gh-8793): Support SQL */
	diag_set(ClientError, ER_CREATE_DEFAULT_FUNC, func_name,
		 "unsupported language");
	return -1;
```